src/Illuminate/Container/Container.php

```diff
@@ -160,7 +161,13 @@ public function bound($abstract)
      */
     public function has($id)
     {
-        return $this->bound($id);
+        try {
+            $this->resolve($id);
```

Comment:
Can't we cache once we resolve this ?

I did prosed an alternative https://github.com/laravel/internals/issues/803#issuecomment-331091077 .